addons/sale/models/sale_order.py

```diff
@@ -1759,11 +1759,7 @@ def show_line(line):
                     # Only show posted down payments
                     or line in down_payment_lines
                 )
-            return (
-                line.display_type == 'line_section'
-                or (line.display_type == 'line_subsection' and not line.parent_id.collapse_composition)
-                or not line.collapse_composition
-            )
```

Comment:
IMO we could keep this, but replace `hide_taxes` to `any(line._has_taxes() for line in lines_to_report`, which would consider child lines recursively?

```python
def _show_taxes(self):
     self.ensure_one()
     return bool(
        self.tax_ids
        or (self.display_type and any(line._show_taxes() for line in self.child_ids))
     )
```

Reply:
Oh, I wanted to say replace `show_taxes`, not `hide_taxes` :sweat_smile: